### DISCUSSION TITLE
Plugin development: ComponentProvider that stacks behind console/bookmark by default

### DISCUSSION THREAD

**Discussion by sad-dev** [NationalSecurityAgency/ghidra#6685] [Q&A] [answered]
I have tried without much success to create a ComponentProvider which is visible by default and that docks with the Console and Bookmark providers.

For example:

```		
setDefaultWindowPosition(WindowPosition.BOTTOM);
setIntraGroupPosition(WindowPosition.BOTTOM);
setVisible(true);
```
just causes my providers to stack at the bottom of the screen. Am I missing a trick here?

**Comment by dragonmacher** [comment, marked as answer]
I believe the issue here is that the Bookmarks and Console providers have been manually grouped together inside of the default `CodeBrowser.tool`.    The call to `setDefaultWindowPosition(WindowPosition.BOTTOM);` will work when creating a new tool from scratch from the UI.  All providers in the BOTTOM group will be together.      Once you have moved providers around and saved your tool, the providers get saved in the updated locations.    The `CodeBrowser.tool` file is a tool config that we created by hand, saved into the repo and  ship in a distribution.  Inside of `CodeBrowser.tool`, these 2 providers have been manually grouped together.

Not all providers we ship are included in the `CodeBrowser.tool` (which may be an oversight).  The `Function Call Trees` provider is one such provider.  It too is set to `setDefaultWindowPosition(WindowPosition.BOTTOM);`.   When it is opened, it appears below the Console and Bookmark providers.

While this may be aesthetically less than desirable, it is only a temporary issue, since the user will place the provider in the preferred location.
